### DISCUSSION TITLE
[Feature Request]Android native invoke

### DISCUSSION THREAD

**Discussion by yymjr** [Tencent/MMKV#1450] [Ideas]
MMKV is a very good library,but I currently find that it cannot be called on Android native.It can only called through Java(athough it can be called using JNI).
But I think since the core is written in C++,why not support it direcylt?

platform:Android

**Comment by lingol** [comment]
It's not by purpose that not support directly called by C++. The problem is that all non-exported symbols are hidden/stripped when we upload it to Maven Central to reduce binary size. 

If you need this functionality, you should build an MMKV native lib by yourself.

**Comment by lingol** [comment]
There are several ways to do it. https://gcc.gnu.org/wiki/Visibility
* Add `__attribute__((visibility("default")))` to the [MMKV](https://github.com/Tencent/MMKV/blob/master/Core/MMKV.h#L96) class declaration (and other classes if you need them).
* Or, remove the argument from [CMake cppFlags](https://github.com/Tencent/MMKV/blob/master/Android/MMKV/mmkv/build.gradle#L25) "-fvisibility=hidden".

**Comment by yymjr** [comment]
You are right,but it seems that prefab can do this.
Build an MMKV native lib by myself? Well, if there is no other way I will do this.

**Comment by lingol** [comment]
We will export symbols and headers in the upcoming big upgrade v2.1.0. https://github.com/Tencent/MMKV/commit/7cad5d3c988fa6f34a7865c94e84df7b283f0309

**Comment by yymjr** [comment]
good
